src/main/java/io/cryostat/jmcagent/JMCAgent.java

```diff
@@ -69,19 +69,24 @@ public Response postProbe(@RestPath long id, @RestPath String probeTemplateName)
             return connectionManager.executeConnectedTask(
                     target,
                     connection -> {
-                        AgentJMXHelper helper = new AgentJMXHelper(connection.getHandle());
-                        String templateContent = service.getTemplateContent(probeTemplateName);
-                        helper.defineEventProbes(templateContent);
-                        ProbeTemplate template = new ProbeTemplate();
-                        template.deserialize(
-                                new ByteArrayInputStream(
-                                        templateContent.getBytes(StandardCharsets.UTF_8)));
-                        bus.publish(
-                                MessagingServer.class.getName(),
-                                new Notification(
-                                        TEMPLATE_APPLIED_CATEGORY,
-                                        Map.of("probeTemplate", template.getFileName())));
-                        return Response.status(RestResponse.Status.OK).build();
+                        try {
+                            AgentJMXHelper helper = new AgentJMXHelper(connection.getHandle());
+                            String templateContent = service.getTemplateContent(probeTemplateName);
+                            helper.defineEventProbes(templateContent);
+                            ProbeTemplate template = new ProbeTemplate();
+                            template.deserialize(
+                                    new ByteArrayInputStream(
+                                            templateContent.getBytes(StandardCharsets.UTF_8)));
+                            bus.publish(
+                                    MessagingServer.class.getName(),
+                                    new Notification(
+                                            TEMPLATE_APPLIED_CATEGORY,
+                                            Map.of("probeTemplate", template.getFileName())));
+                            return Response.status(RestResponse.Status.OK).build();
+                        } catch (Exception e) {
```

Comment:
Would it make sense to add a `helper.defineEventProbes(null)` in this exception handler? This would help in the case I'm seeing where the probe template is valid, but fails at runtime/insertion time.

Reply:
There shouldn't be any residual probes if defining them fails, but it wouldn't hurt to add a cleanup call there.

Reply:
I remembered incorrectly actually https://bugs.openjdk.org/browse/JMC-7949 , this probably explains what you were seeing. I'll find some time later to fix that upstream

Reply:
Yep that looks like it.